publisher-sdk/src/main/java/com/criteo/publisher/Criteo.java

```diff
@@ -32,10 +32,12 @@
 import com.criteo.publisher.util.DeviceUtil;
 import java.util.List;
 
+@Keep
```

Comment:
For your information, `@Keep`ing the whole class means that everything inside is kept. Even private/package elements. Personally, I'm ok to keep them.

This comment is valable for all files

Reply:
Although, you can remove the `@Keep` on the `setUserData`: https://github.com/criteo/android-publisher-sdk/pull/151/files#diff-ed8fa8b89a351ac154181bc558d65aa2a98aa2b3e4f861c13ffd332f5277025dR186